src/core/formState.ts

```diff
@@ -1,23 +1,24 @@
-import { observable, action, computed, runInAction, isObservable, isArrayLike } from 'mobx';
-import { ComposibleValidatable, Validator, applyValidators } from './types';
+import {action, computed, isArrayLike, isObservable, observable, runInAction} from 'mobx';
+import {isES6Map} from "../internal/utils";
+import {applyValidators, ComposibleValidatable, Validator} from './types';
 
 /** Each key of the object is a validatable */
 export type ValidatableMapOrArray =
   { [key: string]: ComposibleValidatable<any> }
-  | ComposibleValidatable<any>[]
+  | ComposibleValidatable<any>[] | Map<any, ComposibleValidatable<any>>
 
 /**
  * Just a wrapper around the helpers for a set of FieldStates or FormStates
  */
 export class FormState<TValue extends ValidatableMapOrArray> implements ComposibleValidatable<TValue> {
-  protected mode: 'array' | 'map' = 'map';
+  protected mode: 'array' | 'map' | 'es6map' = 'map';
   constructor(
     /**
      * SubItems can be any Validatable
      */
     public $: TValue
   ) {
-    this.mode = isArrayLike($) ? 'array' : 'map';
+    this.mode = isArrayLike($) ? 'array' : isES6Map($) ? 'es6map' : 'map';
```

Comment:
Using `isArrayLike` because of mobx wrapping it might not be `Array` and instead be mobx observable array. Same might be true for `Map` i.e. it might not be (at runtime) a true ES6 Map. So perhaps add to `isES6Map` to also use `isObservableMap`.

Reply:
https://mobx.js.org/refguide/api.html